exec/csnex/Object.cs

```diff
@@ -15,30 +49,210 @@ public virtual string toLiteralString()
+                } else {
+                    first = false;
+                }
+                r.Append(Value[x] != null ? Value[x].toLiteralString() : "null");
+            }
+            r.Append("]");
+            return r.ToString();
+        }
+
+        private List<Object> Value;
+    }
+
+    public class ObjectBoolean: Object
+    {
+        public ObjectBoolean(bool b)
+        {
+            Value = b;
+        }
+
+        public ObjectBoolean(ObjectBoolean o)
+        {
+            Value = o.getBoolean(ref Value);
+        }
+
+        public override bool getBoolean(ref bool r)
+        {
+            r = Value;
+            return true;
+        }
+
+        public override string toString()
+        {
+            return Value ? "TRUE" : "FALSE";
+        }
+
+        private bool Value;
+    };
+
+    public class ObjectBytes: Object
+    {
+        public ObjectBytes(byte[] b)
+        {
+            Value = b;
+        }
+
+        public override bool getBytes(ref byte[] b)
+        {
+            b = Value;
+            return true;
+        }
+
+        public override string toString()
+        {
+            StringBuilder r = new StringBuilder("HEXBYTES \"");
+            bool first = true;
+            foreach (byte x in Value) {
+                if (first) {
+                    first = false;
+                } else {
+                    r.Append(' ');
+                }
+                r.AppendFormat("{0:x2}", x);
+            }
+            r.Append("\"");
+            return r.ToString();
+        }
+
+        private byte[] Value;
+    }
+
+    public class ObjectDictionary : Object
+    {
+        public ObjectDictionary(SortedDictionary<string, Object> d)
+        {
+            Value = d;
+        }
+
+        public override bool getDictionary(ref SortedDictionary<string, Object> r)
+        {
+            r = Value;
+            return true;
+        }
+
+        public override bool subscript(Object index, ref Object r)
+        {
+            r = null;
+            string s = null;
+            if (!index.getString(ref s)) {
+                throw new NeonRuntimeException("DynamicConversionException", "to String");
+            }
+            if (s == null) {
+                throw new NeonRuntimeException("DynamicConversionException", "toS tring");
+            }
+
+            if (Value.ContainsKey(s)) {
+                r = Value[s];
+            }
+            if (r == null) {
+                throw new NeonRuntimeException("ObjectSubscriptException", s);
+            }
+            return true;
+        }
+
+        public override string toString()
+        {
+            StringBuilder r = new StringBuilder("{");
+            bool first = true;
+            foreach (KeyValuePair<string, Object> e in Value) {
+                if (first) {
+                    first = false;
+                } else {
+                    r.Append(", ");
+                }
+                r.Append("\"");
```

Comment:
Don't manually quote this here, use your `toLiteralString` or whatever. What if the key value itself contains quotes?

Reply:
Calling `string.Quote()` now instead.